packages/compiler-cli/linker/test/ast/ast_value_spec.ts

```diff
@@ -0,0 +1,314 @@
+      expect(new AstValue(nestedArray, host).getArray()).toEqual([
+        new AstValue(factory.createLiteral(1), host),
+        new AstValue(factory.createLiteral(2), host),
+      ]);
+    });
+
+    it('should throw an error if the property is not an array', () => {
+      expect(() => new AstValue(factory.createLiteral(42), host).getArray())
+          .toThrowError('Unsupported syntax, expected an array literal.');
+    });
+  });
+
+  describe('isFunction', () => {
+    it('should return true if the value is a function expression', () => {
+      const funcExpr = factory.createFunctionExpression(
+          'foo', [],
+          factory.createBlock([factory.createReturnStatement(factory.createLiteral(42))]));
+      expect(new AstValue(funcExpr, host).isFunction()).toEqual(true);
+    });
+
+    it('should return false if the value is not a function expression', () => {
+      expect(new AstValue(factory.createLiteral(42), host).isFunction()).toEqual(false);
+    });
+  });
+
+  describe('getFunctionReturnValue', () => {
+    it('should return the "return value" of the function expression', () => {
+      const funcExpr = factory.createFunctionExpression(
+          'foo', [],
+          factory.createBlock([factory.createReturnStatement(factory.createLiteral(42))]));
+      expect(new AstValue(funcExpr, host).getFunctionReturnValue())
+          .toEqual(new AstValue(factory.createLiteral(42), host));
+    });
+
+    it('should throw an error if the property is not a function expression', () => {
+      expect(() => new AstValue(factory.createLiteral(42), host).getFunctionReturnValue())
+          .toThrowError('Unsupported syntax, expected a function.');
+    });
+
+    it('should throw an error if the property is a function expression with no return value',
+       () => {
+         const funcExpr = factory.createFunctionExpression(
+             'foo', [], factory.createBlock([factory.createExpressionStatement(
+                            factory.createLiteral('do nothing'))]));
+         expect(() => new AstValue(funcExpr, host).getFunctionReturnValue())
+             .toThrowError(
+                 'Unsupported syntax, expected a function body with a single return statement.');
+       });
+  });
+
+  describe('getOpaque()', () => {
+    it('should return the value wrapped in a `WrappedNodeExpr`', () => {
+      expect(new AstValue(factory.createLiteral(42), host).getOpaque())
+          .toEqual(jasmine.any(WrappedNodeExpr));
+      expect(new AstValue(factory.createLiteral(42), host).getOpaque().node)
+          .toEqual(factory.createLiteral(42));
+    });
+  });
+
+  describe('getRange()', () => {
+    it('should return the source range of the AST node', () => {
+      const file = ts.createSourceFile(
+          'test.ts', '// preamble\nx = \'moo\';', ts.ScriptTarget.ES2015,
+          /* setParentNodes */ true);
+      const stmt = file.statements[0] as ts.ExpressionStatement;
+      const expr =
```

Comment:
This would be easier to read with a more specific name:

```suggestion
      const mooString =
```

Reply:
Changed. And I added some comments to make it even clearer.